Documentation/cmdref/cilium-agent.md

```diff
@@ -70,6 +70,7 @@ cilium-agent [flags]
       --config string                                             Configuration file (default "$HOME/ciliumd.yaml")
       --config-dir string                                         Configuration directory that contains a file for each option
       --conntrack-gc-interval duration                            Overwrite the connection-tracking garbage collection interval
+      --controller-group-metrics strings                          List of controller group names for which to to enable metrics. Accepts 'all' and 'none'
```

Comment:
I would consider this a more advanced feature of Cilium, useful in diagnosing failure scenarios for high-volume use cases. Therefore, I think a stability note as you suggested is useful in the flag description and in the Helm documentation.

#### re: The cases where the set of group names changes

##### Group Name is Removed 

If a controller group name is removed, then specifying it in the configuration is a no-op (it's added to the map but never gets hit). If a user was using this metric to monitor that controller group, then the metric is no longer useful, because the controller has been removed or replaced by a new controller. In this case, the removal itself alerts the user that they need to revisit their use of the metric (an alias is not useful for them). Combined with the above stability note, I think that this is reasonable.

##### Group Name is Changed

Similar to the removal case, if a controller group name is changed, then the meaning of the metric has changed as well; something about the implementation of the controller has changed. If a user was using the metric to monitor a controller, then the name change alerts the user that they need to revisit their use of the metric (an alias is not useful for them).

##### Group Name is Added

The most useful documentation for this case would be an exhaustive list of all controller group names available to the Cilium component. However, generating this list is more difficult than it seems, and I would argue that it is not currently worth the effort. 

The `controller.Cell` which enables this feature is made active by this PR for `cilium-agent`, `clustermesh-apiserver`, and `kvstoremesh`. This dynamically includes the metric for all controller groups which are linked-to by that binary, for which there may be overlap between Cilium components. Writing a docsgen which identifies all of the group names present in the binary would be useful, and I think its a good follow-up once we start to use these metrics as operators.

#### re: The default set of group names

Given cardinality concerns, I don't think that defaulting to "all" is the way to go. I would prefer to point users to the default set in the Helm chart.

Given the above, I've added some additional documentation. Do you think it meets our current needs?

Reply:
> Given cardinality concerns, I don't think that defaulting to "all" is the way to go. I would prefer to point users to the default set in the Helm chart.

Agreed, I realized I said `all` in my previous but I will correct that to agree with the default set already provided in the PR.

> Group Name is Changed
> 
> Similar to the removal case, if a controller group name is changed, then the meaning of the metric has changed as well; something about the implementation of the controller has changed. If a user was using the metric to monitor a controller, then the name change alerts the user that they need to revisit their use of the metric (an alias is not useful for them).

That's an important point that I didn't realize. I would agree then that if a metric is "renamed", then it's likely a new metric anyway.